cli/command/container/stats_helpers.go

```diff
@@ -178,10 +178,14 @@ func calculateCPUPercentUnix(previousCPU, previousSystem uint64, v *types.StatsJ
 		cpuDelta = float64(v.CPUStats.CPUUsage.TotalUsage) - float64(previousCPU)
 		// calculate the change for the entire system between readings
 		systemDelta = float64(v.CPUStats.SystemUsage) - float64(previousSystem)
+		onlineCPUs = float64(v.CPUStats.OnlineCPUs)
 	)
 
+	if onlineCPUs == 0.0 {
+		onlineCPUs = float64(len(v.CPUStats.CPUUsage.PercpuUsage))
```

Comment:
This code is on the client side and is to handle the case where the client is talking to an older daemon, so making changes to the daemon side is by definition not an option.

Or perhaps that's not what you meant? I'm not entirely sure where you are coming from, but it seems likely to be relevant to mention that the issue is that the kernel (and hence the daemon) are providing too many entries after the Linux change not too few so adding more dummy entries wouldn't help.